File: GXpert/GXpert.Web/Modules/ServerTypes/Content/ContentRatingForm.ts
import { StringEditor, TextAreaEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface ContentRatingForm {
    Name: StringEditor;
    Description: TextAreaEditor;
}

export class ContentRatingForm extends PrefixedContext {
    static readonly formKey = 'Content.ContentRating';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ContentRatingForm.init)  {
            ContentRatingForm.init = true;

            var w0 = StringEditor;
            var w1 = TextAreaEditor;

            initFormType(ContentRatingForm, [
                'Name', w0,
                'Description', w1
            ]);
        }
    }
}

File: GXpert/GXpert.Web/Modules/Content/ContentRating/ContentRatingForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Content.Forms;

[FormScript("Content.ContentRating")]
[BasedOnRow(typeof(ContentRatingRow), CheckNames = true)]
public class ContentRatingForm
{
    public string Name { get; set; }
    [TextAreaEditor(Rows =3)]
    public string Description { get; set; }
   
}